core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1606,7 +1608,7 @@ void renameInternal(LockedInodePath srcInodePath, LockedInodePath dstInodePath,
       UnderFileSystem ufs = resolution.getUfs();
       String ufsDstUri = mMountTable.resolve(dstPath).getUri().toString();
       String parentUri = new AlluxioURI(ufsDstUri).getParent().toString();
-      if (!ufs.exists(parentUri)) {
+      if (!ufs.isDirectory(parentUri)) {
```

Comment:
If the uri is a file, the following mkdirs operation will return false or through an exception (the UFSs may not be consistent). The end result will be in both cases = renameInternal throws an IOException which is the correct behavior. 

Earlier if parentUri was a file, the following rename would fail and through an IOException. The difference is in where renameInternal fails but the end result will be the same.